### PR TITLE
Clustering API

### PR DESCRIPTION
**Pull request by sumerjabri**
### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/2583

Added clustering API. Removed password field from AuthenticatedUser, and added notes for password fields so they're not returned to the caller.

### CODE REVIEW

File: src/main/api/2/studio-api-2.yaml [hunk 1]
```diff
@@ -1115,6 +1119,159 @@ paths:
...
+                      entities:
+                        type: array
+                        items:
+                          $ref: '#/components/schemas/ClusterMember'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
```
Comment (avasquez614) [inline src/main/api/2/studio-api-2.yaml:1148]: I don't think we need to throw this error (404)?

File: src/main/api/2/studio-api-2.yaml [hunk 2]
```diff
@@ -1115,6 +1119,159 @@ paths:
...
+      operationId: updateClusterMembers
+      requestBody:
+        description: Cluster members to update (upsert)
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                clusterMembers:
```
Comment (avasquez614) [inline src/main/api/2/studio-api-2.yaml]: We're not wrapping currently the request objects. I would say remove `clusterMembers`

File: src/main/api/2/studio-api-2.yaml [hunk 3]
```diff
@@ -1115,6 +1119,159 @@ paths:
...
+                        $ref: '#/components/schemas/ApiResponse'
+                      entities:
+                        type: array
+                        items:
+                          $ref: '#/components/schemas/ClusterMember'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
```
Comment (avasquez614) [inline src/main/api/2/studio-api-2.yaml]: I don't think we need to throw this error (404)?

**Review by avasquez614 — DISMISSED** [review]

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (sumerjabri) [reply src/main/api/2/studio-api-2.yaml:1148]: What should we return if there is no cluster (single node with the DB table empty)?

Comment (avasquez614) [reply src/main/api/2/studio-api-2.yaml:1148]: You're right we should return 404

**Pull request merged**